az-digital/az_quickstart
# Fixes #2255, Fixes #2299 API Endpoint for News Export

## Description
This PR adds a module that contains a REST endpoint view for exporting News content to outside services.

There are two primary potential approaches this could be done with; REST and JSON:API. JSON:API is easiest, but because this data involves multiple chained references, the JSON:API output is not very easy to work with on the receiving side. To that end, this draft is attempting the REST version first, to provide output that is more directly structured.

This PR contains a view and a Serializer REST plugin that implement changing some of the references fields in the output to a more streamlined form. This approach is needed because adding true views relationships results in duplicate rows (and also doesn't provide an array structure that makes for simple output.

The JSON:API approach could be swapped to if desirable.

## Related issues
#2255 
#2299 

## How to test
- Enable `Quickstart News Export`
- Generate some News Content
- verify JSON output at `/az_quickstart/1.0/export/az_news`

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
